swarm/swap/swap.go

```diff
@@ -0,0 +1,589 @@
+	cheque := sp.swapAccount.chequeManager.CreateCheque(sp.ID(), amount.Abs(payAt))
+	msg := IssueChequeMsg{
+		Cheque: cheque,
+	}
+	return sp.Send(ctx, msg)
+}
+
+//Create a new swap accounted peer
+func NewSwapPeer(peer *protocols.Peer, swap *Swap) *SwapPeer {
+	balance := big.NewInt(0)
+	//check if there is one already in the stateStore and load it
+	swap.stateStore.Get(peer.String()[:24]+"-swap", &balance)
+	sp := &SwapPeer{
+		Peer:        peer,
+		swapAccount: swap,
+		balance:     balance,
+		storeID:     peer.String()[:24] + "-swap",
+	}
+	swap.lock.Lock()
+	defer swap.lock.Unlock()
+	swap.peers[peer.ID()] = sp
+	return sp
+}
+
+// Profile - public swap profile
+// public parameters for SWAP, serializable config struct passed in handshake
+type Profile struct {
+	BuyAt  *big.Int // accepted max price for chunk
+	SellAt *big.Int // offered sale price for chunk
+	PayAt  *big.Int // threshold that triggers payment request
+	DropAt *big.Int // threshold that triggers disconnect
+}
+
+// Strategy encapsulates parameters relating to
+// automatic deposit and automatic cashing
+type Strategy struct {
+	AutoCashInterval     time.Duration // default interval for autocash
+	AutoCashThreshold    *big.Int      // threshold that triggers autocash (wei)
+	AutoDepositInterval  time.Duration // default interval for autocash
+	AutoDepositThreshold *big.Int      // threshold that triggers autodeposit (wei)
+	AutoDepositBuffer    *big.Int      // buffer that is surplus for fork protection etc (wei)
+}
+
+// SwapMsg encapsulates messages transported over pss.
+type SwapMsg struct {
+	To      []byte
+	Control []byte
+	Expire  uint32
+	Payload *whisper.Envelope
+}
+
+// Params extends the public profile with private parameters relating to
+// automatic deposit and automatic cashing
+type Params struct {
+	*Profile
+	*Strategy
+}
+
+// LocalProfile combines a PayProfile with *swap.Params
+type LocalProfile struct {
+	*Params
+	*PayProfile
+}
+
+// RemoteProfile combines a PayProfile with *swap.Profile
+type RemoteProfile struct {
+	*Profile
+	*PayProfile
+}
+
+// PayProfile is a container for relevant chequebook and beneficiary options
+type PayProfile struct {
+	PublicKey   string         // check against signature of promise
+	Contract    common.Address // address of chequebook contract
+	Beneficiary common.Address // recipient address for swarm sales revenue
+	privateKey  *ecdsa.PrivateKey
+	publicKey   *ecdsa.PublicKey
+	owner       common.Address
+	chbook      *chequebook.Chequebook
+	lock        sync.RWMutex
+}
+
+// New - swap constructor
+func NewSwap(local *Params, stateStore state.Store) (swap *Swap, err error) {
+
+	swap = &Swap{
+		chequeManager: NewChequeManager(stateStore),
+		local:         local,
+		stateStore:    stateStore,
+		peers:         make(map[discover.NodeID]*SwapPeer),
+	}
+
+	//swap.SetParams(local)
+	return
+}
+
+// NewDefaultSwapParams create params with default values
+func NewDefaultSwapParams() *LocalProfile {
```

Comment:
i would really prefer  not migrating this yet, already too  much for this PR ,no?